server/src/main/java/org/elasticsearch/cluster/health/ClusterStateHealth.java

```diff
@@ -134,6 +137,11 @@ public ClusterStateHealth(final StreamInput in) throws IOException {
         unassignedShards = in.readVInt();
         numberOfNodes = in.readVInt();
         numberOfDataNodes = in.readVInt();
+        if (in.getVersion().onOrAfter(Version.V_7_0_0)) {
```

Comment:
By the time this lands, `master` will be `V_8_0_0`, and you will have to do the BWC dance to backport it to `7.x` and `7.0`. I also think this counts as a feature and therefore it shouldn't be backported to `7.0`?

The same comment applies to other places that look like this.

Reply:
Ok, I have done the following things:

1) Repurposed the PR to 8.0 and 7.1
2) Disabled bwc tests for PR that goes to 8.0
3) Added TODO to change V_7_0_0 once https://github.com/elastic/elasticsearch/pull/38513 is merged

Reply:
Now because #38513 is merged, I've changed the version to V_7_1_0